### ISSUE TITLE
RSA decoding crash

### ISSUE THREAD

**Issue by AgeManning** [libp2p/rust-libp2p#1173]
We've started doing some fuzz testing on the protocols we're building into libp2p. 

One test demonstrates that RSA decoding can cause memory crashes (tries to allocate an unreasonably large vec). It can be reproduced by adding the the following test to `/core/src/identity/rsa.rs`:
```rust
    #[test]
    fn rsa_x509_decode_bytes() {
        let bytes = [
            157, 247, 157, 157, 157, 157, 157, 157, 157, 157, 157, 157, 157, 157, 157, 67, 157, 1,
            0, 0, 0, 157, 157, 157, 157, 157, 157, 157, 157,
        ];
        let _ = PublicKey::decode_x509(&bytes);
    }
```
This is an issue with the `ans1_der` library, not libp2p. Anyway, if there are protocols that decode public keys from external sources, be aware of the above.

This is related to: https://github.com/KizzyCode/asn1_der/issues/1

**Comment by tomaka** [comment]
It might be a good idea to allow disabling RSA at compile-time with a feature.

**Comment by KizzyCode** [comment]
Ok, the bug in [`asn1_der`](https://github.com/KizzyCode/asn1_der/issues/1) has been fixed in `0.6.2`; thank you for raising this issue 😊

**Comment by AgeManning** [comment]
This has been fixed :). Thanks @KizzyCode
